.specs/design.md

```
@@ -0,0 +1,921 @@
+  FOREIGN KEY (server_id) REFERENCES servers(id) ON DELETE CASCADE
+);
+
+-- Indexes for performance optimization
+CREATE INDEX idx_configurations_server_id ON configurations(server_id);
+CREATE INDEX idx_configurations_environment ON configurations(environment);
+CREATE INDEX idx_configurations_enabled_tools ON configurations(enabled_tools);
+CREATE INDEX idx_configurations_resource_limits ON configurations(resource_limits);
+CREATE INDEX idx_configurations_network_config ON configurations(network_config);
+```
+);
+```
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)